packages/vite/src/node/server/mixedModuleGraph.ts

```diff
@@ -308,7 +308,7 @@ export class ModuleGraph {
     if (ssrModules) {
       for (const mod of ssrModules) {
         if (mod.id == null || !this._client.getModuleById(mod.id)) {
-          result.add(this.getBackwardCompatibleBrowserModuleNode(mod)!)
+          result.add(this.getBackwardCompatibleServerModuleNode(mod)!)
```

Comment:
When `getBackwardCompatibleBrowserModuleNode` is used, a ssr module gets `._clientModule` and that will be included in the client env hmr module list by
https://github.com/vitejs/vite/blob/fd1283fe27cc1a19b5c7d9d72664832e4daa1bbf/packages/vite/src/node/server/hmr.ts#L517-L527